framework/i18n/MessageFormatter.php

```diff
@@ -274,19 +275,20 @@ protected function fallbackFormat($pattern, $args, $locale)
      */
     private static function tokenizePattern($pattern)
     {
+        $charset = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
I thinks this is fine, if you choose another encoding for whatever reason, all strings should have the same encoding.